### PR TITLE
gh-109435: Add Doc/library/cmdline.rst

### PR DESCRIPTION
**Pull request by vstinner**
Document modules providing a command-line interface (CLI).

* Issue: gh-109435

----
:books: Documentation preview :books:: https://cpython-previews--109436.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/cmdline.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+The following modules have a command-line interface.
+
+* :ref:`ast <ast-cli>`
+* :ref:`asyncio <asyncio-cli>`
+* :mod:`base64`
+* :ref:`calendar <calendar-cli>`
+* :mod:`cProfile`: see :ref:`profile <profile-cli>`
+* :mod:`code`
+* :ref:`compileall <compileall-cli>`
```
Comment (hugovk) [inline Doc/library/cmdline.rst]:
```suggestion
* :mod:`code`
* :ref:`compileall <compileall-cli>`
* :mod:`cProfile`: see :ref:`profile <profile-cli>`
```

File: Doc/library/cmdline.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+* :ref:`inspect <inspect-module-cli>`
+* :ref:`json.tool <json-commandline>`
+* :mod:`mimetypes`
+* :mod:`pdb`
+* :mod:`pickle`
+* :ref:`pickletools <pickletools-cli>`
+* :mod:`platform`
+* :mod:`poplib`
+* :mod:`pstats`
+* :ref:`profile <profile-cli>`
```
Comment (hugovk) [inline Doc/library/cmdline.rst]:
```suggestion
* :ref:`profile <profile-cli>`
* :mod:`pstats`
```

File: Doc/library/cmdline.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* :ref:`sysconfig <sysconfig-cli>`
+* :mod:`tabnanny`
+* :ref:`tarfile <tarfile-commandline>`
+* :mod:`this`
+* :ref:`timeit <timeit-command-line-interface>`
+* :ref:`tokenize <tokenize-cli>`
+* :ref:`trace <trace-cli>`
+* :mod:`turtledemo`
+* :ref:`uuid <uuid-cli>`
+* :ref:`unittest <unittest-command-line-interface>`
```
Comment (hugovk) [inline Doc/library/cmdline.rst]:
```suggestion
* :ref:`unittest <unittest-command-line-interface>`
* :ref:`uuid <uuid-cli>`
```

**Review by hugovk — COMMENTED** [review]
Sort alphabetically.

**Comment by vstinner** [comment]
@hugovk @AA-Turner: Would you mind to review my updated/completed doc?

Some people proposed to automate the generation of this list, but it can be done later, no?

**Review by hugovk — APPROVED** [review]
👍

**Comment by vstinner** [comment]
PR rebased on main to try to get green CI jobs :-)

**Comment by vstinner** [comment]
Merged. Thanks for the review @hugovk.

File: Doc/library/cmdline.rst
Comment (AA-Turner) [inline Doc/library/cmdline.rst:1]: Sorry for being late to re-reivew here; I still think we should at least have grouped these entries (https://github.com/python/cpython/pull/93410#issuecomment-1180129323), per the discussion in Discord.

A

Comment (vstinner) [reply Doc/library/cmdline.rst:1]: Feel free to propose a follow-up PR to enhance this list.

**Pull request merged**